### PR TITLE
Add unity asset loading example

### PR DESCRIPTION
**Pull request by FredrikNoren**

### CODE REVIEW

File: guest/rust/examples/assets/unity/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+[package]
+name = "unity"
+
+edition = "2021"
+publish = false
+rust-version = { workspace = true }
+version = { workspace = true }
+
+[dependencies]
+ambient_api = { path = "../../../api" }
```
Comment (philpax) [inline guest/rust/examples/assets/unity/Cargo.toml:10]: workspace?

File: guest/rust/examples/assets/unity/download.sh [hunk 2]
```diff
@@ -0,0 +1,37 @@
+export BASE="${EXAMPLES_ASSETS_HOST}/unity/Dynamic%20Nature%20-%20Mountain%20Tree%20Pack"
```
Comment (philpax) [inline guest/rust/examples/assets/unity/download.sh:1]: Is EXAMPLES_ASSETS_HOST defined somewhere?

Comment (FredrikNoren) [reply guest/rust/examples/assets/unity/download.sh:1]: Yeah it's a github secret, because it's not public content

Comment (FredrikNoren) [reply guest/rust/examples/assets/unity/download.sh:1]: (In the future we should switch to public content, but to get started I wanted to use something I know at least works somewhat)

**Pull request merged**